### PR TITLE
feat: CDN Updates

### PR DESCRIPTION
**Pull request by bkochendorfer**
This adds some CDN configuration options that I will need to implement the Bedrock CDN using this shared module. I also added in this document yaml to regenerate. Hopefully that is the right thing or I can back out adding that.

### CODE REVIEW

File: google_cdn-external/variables.tf [hunk 1]
```diff
@@ -115,3 +115,15 @@ variable "quic_override" {
...
+variable "custom_response_headers" {
+  description = "Specifies custom response headers"
+  type        = list(string)
+  default     = null
+}
+
+variable "custom_request_headers" {
+  description = "Specifies custom response headers"
+  type        = list(string)
+  default     = null
```
Comment (jasonthomas) [inline google_cdn-external/variables.tf:128]:
```suggestion
  default     = []
```
concat doesn't seem to evaluate null properly
```
> concat(["test"], null)

> ╷
│ Error: Invalid function argument
│
│   on <console-input> line 1:
│   (source code not available)
│
│ Invalid value for "seqs" parameter: argument must not be null.
╵

> concat(["test"], [])
[
  "test",
]
```

File: google_cdn-external/variables.tf [hunk 2]
```diff
@@ -115,3 +115,15 @@ variable "quic_override" {
   type        = string
   default     = "DISABLE"
 }
+
+variable "custom_response_headers" {
+  description = "Specifies custom response headers"
+  type        = list(string)
+  default     = null
```
Comment (jasonthomas) [inline google_cdn-external/variables.tf:122]:
```suggestion
  default     = []
```
Probably can also change it here too.

File: google_cdn-external/variables.tf (hunk 1 above)
Comment (bkochendorfer) [reply google_cdn-external/variables.tf:128]: That is a great catch @jasonthomas

**Review by jasonthomas — APPROVED** [review]

File: google_cdn-external/main.tf [hunk 3]
```diff
@@ -55,9 +55,13 @@ resource "google_compute_backend_service" "default" {
       serve_while_stale            = lookup(var.cdn_policy, "serve_while_stale", null)
       signed_url_cache_max_age_sec = lookup(var.cdn_policy, "signed_url_cache_max_age_sec", null)
       cache_key_policy {
-        include_host         = lookup(var.cache_key_policy, "include_host", true)
-        include_protocol     = lookup(var.cache_key_policy, "include_protocol", true)
-        include_query_string = lookup(var.cache_key_policy, "include_query_string", true)
+        include_host           = lookup(var.cache_key_policy, "include_host", true)
+        include_protocol       = lookup(var.cache_key_policy, "include_protocol", true)
+        include_query_string   = lookup(var.cache_key_policy, "include_query_string", true)
+        include_http_headers   = lookup(var.cache_key_policy, "include_https_heaers", null)
```
Comment (cvalaas) [inline google_cdn-external/main.tf:61]: typo on this line?

**Review by cvalaas — CHANGES_REQUESTED** [review]

File: google_cdn-external/main.tf (hunk 3 above)
Comment (cvalaas) [reply google_cdn-external/main.tf:61]: "heaers"

Comment (cvalaas) [reply google_cdn-external/main.tf:61]: also the var is include_HTTP_headers while the lookup is for include_HTTPS_...

Comment (cvalaas) [reply google_cdn-external/main.tf:61]: (maybe that last one is intentional?)

Comment (bkochendorfer) [reply google_cdn-external/main.tf]: Just a typo! Thanks for catching

**Review by cvalaas — DISMISSED** [review]

**Comment by bkochendorfer** [comment]
I added a couple more things here to get this to work with cert-manager. I'm still waiting to see if I can move forward with this option for my project in GCP.

**Review by jasonthomas — APPROVED** [review]

**Comment by bkochendorfer** [comment]
@jasonthomas I ended up not being able to use this module and maybe also not use the CDN haha. I think I'll close this for now, we can always resurrect if needed.

**Pull request closed without merging**